app/src/main/java/com/asma/tasky/di/AuthModule.kt

```diff
@@ -0,0 +1,39 @@
+package com.asma.tasky.di
+
+import android.content.SharedPreferences
+import com.asma.tasky.feature_authentication.data.remote.AuthenticationApi
+import com.asma.tasky.feature_authentication.data.repository.AuthenticationRepositoryImpl
+import com.asma.tasky.feature_authentication.domain.repository.AuthenticationRepository
+import dagger.Module
+import dagger.Provides
+import dagger.hilt.InstallIn
+import dagger.hilt.components.SingletonComponent
+import okhttp3.OkHttpClient
+import retrofit2.Retrofit
+import retrofit2.converter.gson.GsonConverterFactory
+import javax.inject.Singleton
+
+@Module
+@InstallIn(SingletonComponent::class)
+object AuthModule {
+
+    @Provides
+    @Singleton
+    fun provideAuthApi(client: OkHttpClient): AuthenticationApi {
+        return Retrofit.Builder()
+            .baseUrl(AuthenticationApi.BASE_URL)
+            .client(client)
+            .addConverterFactory(GsonConverterFactory.create())
```

Comment:
I'd recommend to use anything but Gson for JSON parsing. Gson is a Java library and has its issues with things like nullability and default values in Kotlin (for example, it can assign null values to nullable fields). Especially for Retrofit it's very easy to swap it out with a Kotlin library like Moshi or Kotlin serialization 😄